jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/mapper/PostMapperDecorator.java

```diff
@@ -0,0 +1,52 @@
+package com.example.learning.mapper;
+
+import com.example.learning.entities.Post;
+import com.example.learning.entities.Tag;
+import com.example.learning.model.request.PostCommentRequest;
+import com.example.learning.model.request.PostRequest;
+import com.example.learning.model.request.TagRequest;
+import com.example.learning.repository.TagRepository;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class PostMapperDecorator implements PostMapper {
+
+    @Autowired
+    private TagRepository tagRepository;
```

Comment:
@coderabbitai, how to use constructor injection instead of field injection in the Decorator of PostMapper defined using `MapStruct`

Reply:
@coderabbitai, receiving error as below, please fix

Specified decorator type has no default constructor nor a constructor with a single parameter accepting the decorated mapper type.